common/containers/Tabs/SendTransaction/index.jsx

```diff
@@ -334,44 +338,33 @@ export class SendTransaction extends React.Component {
     );
   }
 
-  async getTransactionInfoFromState(): Promise<TransactionWithoutGas> {
+  async getFormattedTxFromState(): Promise<TransactionWithoutGas> {
     const { wallet } = this.props;
     const { token, unit, value, to, data } = this.state;
-
-    if (unit === 'ether') {
-      return {
-        to,
-        from: await wallet.getAddress(),
-        value: valueToHex(value),
-        data
-      };
-    } else {
-      if (!token) {
-        throw new Error('No matching token');
-      }
-
-      const bigAmount = new Big(value);
-
-      return {
-        to: token.address,
-        from: await wallet.getAddress(),
-        value: '0x0',
-        data: ERC20.transfer(to, toTokenUnit(bigAmount, token))
-      };
-    }
+    const transactionInput: TransactionInput = {
+      token,
+      unit,
+      value,
+      to,
+      data
+    };
+    return await formatTxInput(wallet, transactionInput);
   }
 
   async estimateGas() {
     if (!isNaN(parseInt(this.state.value))) {
```

Comment:
Better if you check the negation and break out of the function early if so, saves you a nested block